### ISSUE TITLE
Supporting  ecs-logging-java

### ISSUE THREAD

**Issue by thaarbach** [rfoltyns/log4j2-elasticsearch#55]
### Feature Request
Elastic has created a uniform scheme for the elastic stack. For logging with Log4j2 the log4j2-ecs-layout has been published.

This feature would also make the log4j2 appenders compatible to Log-Ui in Kibana. 

See
[ECS-based logging for Java applications](https://github.com/elastic/ecs-logging-java)
[Log4j2 ECS Layout](https://github.com/elastic/ecs-logging-java/blob/master/log4j2-ecs-layout/README.md)

Example:
```
<Appenders>
    <Elasticsearch name="elasticsearchAsyncBatch">
        <IndexName indexName="log4j2"/>
        <EcsLayout>
                <KeyValuePair key="additionalField1" value="constant value"/>
                <KeyValuePair key="additionalField2" value="$${ctx:key}"/>
         </EcsLayout>
        <AsyncBatchDelivery batchSize="1000" deliveryInterval="10000" >
            <HCHttp serverUris="http://localhost:9200">
                <PooledItemSourceFactory poolName="batchPool" itemSizeInBytes="1024000" initialPoolSize="3"/>
            </HCHttp>
        </AsyncBatchDelivery>
    </Elasticsearch>
</Appenders>
```

This could make also `<IndexTemplate name="log4j2" path="classpath:indexTemplate-7.json" />` obsolet.

**Comment by rfoltyns** [comment]
Thanks, @thaarbach. It makes sense to get closer to ECS, but I'm not planning to support `ECSLayout` directly. It should be possible to get there with new JacksonMixIn.

**Comment by thaarbach** [comment]
@rfoltyns whould be nice if there is way to support it. We switched to elastic-apm and it would be a great think to use the LogUi.

**Comment by rfoltyns** [comment]
Absolutely! :) `JacksonJsonLayout` and mixin-based approach should be easy to achieve.

Actually, `EcsLayout` can be used with `JestHttp` as-is. No pooling is possible though, so it has several times higher memory allocation rate than `log4j2-elasticsearch-hc`.

Once the mixin is ready, I'll release it in 1.4.5, cause 1.5 will take at least few more weeks.

**Comment by rfoltyns** [comment]
I've created a simple mixin for it. @thaarbach would you like to test it?

**Comment by thaarbach** [comment]
@rfoltyns sorry, was very busy. I will test it next week.

**Comment by rfoltyns** [comment]
I'll have to release next version over the weekend.

I'll include it and add eventual patches later if needed.

**Comment by rfoltyns** [comment]
@thaarbach Released in 1.4.5. Feel free to re-open if you'd like to see more changes in this scope.

**Comment by thaarbach** [comment]
@rfoltyns  
Hi Rafal,
Thx for implementing that feature. :+1:
